assignment5/assignment5.cpp

```diff
@@ -12,33 +12,45 @@ class UnknownLanguage{
 	std::vector<std::string> dict;
 	void DictionaryToGraph(){
 		graph.clear();
-		for(int i=0; i<dict.size()-1; i++){
-			for(int j=0; j<dict[i].size(); j++){
-
-				if(j < dict[i+1].size() && 
-					dict[i].substr(0, j) == dict[i+1].substr(0, j)){
-
+		bool skip = false;
+		for(int i = 0; i+1<dict.size(); i++){
+			skip = false;
+			for(int j = 0; j<dict[i].size(); j++){
+				if(!skip && j < dict[i+1].size()){
 					if(dict[i][j]!=dict[i+1][j]){
 						graph[dict[i][j]].out_edges.push_back(dict[i+1][j]);
 						graph[dict[i+1][j]].in_edges.push_back(dict[i][j]);
+						skip = true;
 					}
 				} else {
-					break;
+					skip = true;
+					if (graph.find(dict[i][j])==graph.end()){
+						graph.insert(std::pair<char,Node>(dict[i][j],Node()));
+					}
+					
 				}
 			}
 
 		}
+
+		for(int i = 0; i < dict.back().size(); i++){
```

Comment:
I think code will look cleaner if you split it in two blocks: first you compute the set of all letters in the alphabet generating a entry pointing to a empty `Node()` in `graph`. Then compute the edges.